### ISSUE TITLE
Translation to Persian

### ISSUE THREAD

**Issue by jowharshamshiri** [huggingface/course#50]
Hi there 👋

Let's translate the course to Persian so that the whole community can benefit from this resource 🌎!

Below are the chapters and files that need translating - let us know here if you'd like to translate any and we'll add your name to the list. Once you're finished, open a pull request and tag this issue by including `#issue-number` in the description, where `issue-number` is the number of this issue.

> 🙋 If you'd like others to help you with the translation, you can also post in our [forums](https://discuss.huggingface.co/c/course/20) or tag [@_lewtun](https://twitter.com/_lewtun) on Twitter to gain some visibility.

## Chapters

**0 - Setup**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter0/1.mdx)

**1 - Transformer models**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/1.mdx)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/3.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/4.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/6.mdx)
- [ ] [`7.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/7.mdx)
- [ ] [`8.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/8.mdx)
- [ ] [`9.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/9.mdx)
- [ ] [`10.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/10.mdx)

**2 - Using 🤗 Transformers**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/1.mdx)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/3.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/4.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/6.mdx)
- [ ] [`7.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/7.mdx)
- [ ] [`8.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/8.mdx)

**3 - Fine-tuning a pretrained model**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/1.mdx)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/3.mdx)
- [ ] [`3_tf.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/3_tf.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/4.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/6.mdx)

**4 - Sharing models and tokenizers**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter4/1.mdx)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter4/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter4/3.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter4/4.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter4/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter4/6.mdx)

**5 - The 🤗 Datasets library**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/1.mdx)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/3.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/4.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/6.mdx)
- [ ] [`7.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/7.mdx)
- [ ] [`8.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/8.mdx)

**6 -  The 🤗 Tokenizers library**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/1.mdx)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/3.mdx)
- [ ] [`3b.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/3b.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/4.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/6.mdx)
- [ ] [`7.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/7.mdx)
- [ ] [`8.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/8.mdx)
- [ ] [`9.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/9.mdx)
- [ ] [`10.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/10.mdx)

**7 - Main NLP tasks**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/1.mdx)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/3.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/4.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/6.mdx)
- [ ] [`7.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/7.mdx)
- [ ] [`8.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/8.mdx)
- [ ] [`9.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/9.mdx)

**8 - How to ask for help**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/1.mdx)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/3.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/4.mdx)
- [ ] [`4_tf.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/4_tf.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/6.mdx)
- [ ] [`7.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/7.mdx)

**Events**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/event/1.mdx)

**Comment by schoobani** [comment]
@jowharshamshiri **per**, **fas** and **fa** can be language codes for persian. I suggest using **fa**.

**Comment by schoobani** [comment]
I am also delighted to contribute.

I forked and added the fa directory and the `_toctree.yml` file. you can track my updates in here:
https://github.com/schoobani/course/tree/main/chapters/fa

**Comment by jowharshamshiri** [comment]
Awesome, thanks!

**Comment by lewtun** [comment]
Hi @schoobani and @jowharshamshiri thank you so much for this initiative and for offering to help! I've added your name @schoobani to the Chapter 1 list. In case it's easier, you don't have to translate the whole chapter in one go - feel free to just add the sections as you work through them!

**Comment by jowharshamshiri** [comment]
@schoobani I propose we use this Persian style guide to ensure uniform style across all of our work. Looks intimidating but has has a lot of whitespace and is the best I could find. You don't need to read it, just refer to it for common Persian writing guidelines like prefixes and suffixes and spacing and such. 
https://apll.ir/wp-content/uploads/2018/10/D-1394.pdf

**Comment by jowharshamshiri** [comment]
@schoobani I propose that in Persian text we replace the use of the Huggingface emoji with the transliteration of the word 'Huggingface' in Persian(هاگینگ‌فیس). What do you think? Emoji's don't lend themselves easily to Persian text. @lewtun I don't know if that is against Huggingface policy, love to hear your thoughts on this.

**Comment by jowharshamshiri** [comment]
@lewtun I added my own name to the Chapter 2 list.

**Comment by lewtun** [comment]
Hello @schoobani has a nice PR here: https://github.com/huggingface/course/pull/71

Would a Persian speaker here be willing to have a quick review as I unfortunately don't speak Farsi 🙈 ?

**Comment by schoobani** [comment]
> @schoobani I propose that in Persian text we replace the use of the Huggingface emoji with the transliteration of the word 'Huggingface' in Persian(هاگینگ‌فیس). What do you think? Emoji's don't lend themselves easily to Persian text. @lewtun I don't know if that is against Huggingface policy, love to hear your thoughts on this.

@jowharshamshiri I think using the emoji 🤗 is compatible with the Persian text. Personally for me, seeing the emoji in documents indicates a relation with huggingface features. We can decide about it after finalizing the translations.

**Comment by jowharshamshiri** [comment]
> > @schoobani I propose that in Persian text we replace the use of the Huggingface emoji with the transliteration of the word 'Huggingface' in Persian(هاگینگ‌فیس). What do you think? Emoji's don't lend themselves easily to Persian text. @lewtun I don't know if that is against Huggingface policy, love to hear your thoughts on this.
> 
> @jowharshamshiri I think using the emoji hugs is compatible with the Persian text. Personally for me, seeing the emoji in documents indicates a relation with huggingface features. We can decide about it after finalizing the translations.

I think we should make these choices as soon as possible, to minimize editing at the end. This one seems like a simple replace, but there will be so many little details like it that when taken together, will make for a quality translation and let us avoid a lot of unnecessary headache. That being said, these are my reasons for transliteration:

- This is a substitute for brand name. We should transliterate brand names, except when we have a specific reason not to. I do think your argument is valid that this is an immediate visual reference to a Hugging Face-related feature, but is this reason enough? I don't think so.
- Mixing non-Persian characters into Persian makes for a fragmentary and incomprehensible text. It should be avoided where at all possible and tolerated where necessary. Programming namespaces are one example that we have to tolerate. The effect might not be noticeable in this small choice, but I promise you is very noticeable for a new reader. It's like when I look at a painting and can't really describe why I like it. Art historians can put words to my feelings, but I do feel them nevertheless. Persian text should flow naturally--this breaks that flow. 

Also I went with 'Huggingface'(one word) to demonstrate our use of a half-space in Persian transliteration to Lewton. The English is ofc Hugging Face(two words) but when transliterated to Persian, this is one semantic unit and a loan word that we are introducing to Persian, which is reflected in the half-space. You don't expect people to use each transliterated word elsewhere in Persian on its own.

Happy to hear your thoughts on this.

**Comment by jowharshamshiri** [comment]
> > @schoobani I propose that in Persian text we replace the use of the Huggingface emoji with the transliteration of the word 'Huggingface' in Persian(هاگینگ‌فیس). What do you think? Emoji's don't lend themselves easily to Persian text. @lewtun I don't know if that is against Huggingface policy, love to hear your thoughts on this.
> 
> Thanks for raising this interesting point! If emoji's aren't a good fit for Persian then I think it's perfectly fine to transliterate "Hugging Face" or "Huggingface" if the latter is easier :)

I went with 'Huggingface'(one word) to demonstrate our use of a half-space in Persian transliteration to you. The English is ofc Hugging Face(two words) but when transliterated to Persian, this is one semantic unit and a loan word that we are introducing to Persian, which is reflected in the half-space. You don't expect people to use each transliterated word elsewhere in Persian on its own.

**Comment by schoobani** [comment]
> 
Sure, we should use the single word structure (هاگینگ‌فیس), and I will consider it on my translations.
We must consider the correctly usage of half-space as a general rule in all translated documents not just the name.

**Comment by schoobani** [comment]
> > > @schoobani I propose that in Persian text we replace the use of the Huggingface emoji with the transliteration of the word 'Huggingface' in Persian(هاگینگ‌فیس). What do you think? Emoji's don't lend themselves easily to Persian text. @lewtun I don't know if that is against Huggingface policy, love to hear your thoughts on this.
> > 
> > 
> > Thanks for raising this interesting point! If emoji's aren't a good fit for Persian then I think it's perfectly fine to transliterate "Hugging Face" or "Huggingface" if the latter is easier :)
> 
> I went with 'Huggingface'(one word) to demonstrate our use of a half-space in Persian transliteration to you. The English is ofc Hugging Face(two words) but when transliterated to Persian, this is one semantic unit and a loan word that we are introducing to Persian, which is reflected in the half-space. You don't expect people to use each transliterated word elsewhere in Persian on its own.

Sure, we should use the single word structure, and I will consider it on my translations.
We must consider the correctly usage of half-space as a general rule in all translated documents not just the name.

**Comment by jowharshamshiri** [comment]
@lewtun I added my name to chapter 0 and also chapter 2/page 1.

**Comment by schoobani** [comment]
@jowharshamshiri I mistakenly translated the [chapter 0](https://github.com/schoobani/course/blob/main/chapters/fa/chapter0/1.mdx). If you have already done it feel free to ignore my work. As soon as you let me know I send a pull request. so we can see the first previews of RTL format on huggingface docs.

**Comment by schoobani** [comment]
@jowharshamshiri I am not sure how the translated numbers will be shown in the docs and which is the best approach. 
```
- title: 0. نصب
  sections:
  - local: chapter0/1
    title: مقدمه
```

Or?

```
- title: ۰. نصب
  sections:
  - local: chapter0/1
    title: مقدمه
```

**Comment by jowharshamshiri** [comment]
> @jowharshamshiri I mistakenly translated the [chapter 0](https://github.com/schoobani/course/blob/main/chapters/fa/chapter0/1.mdx). If you have already done it feel free to ignore my work. As soon as you let me know I send a pull request. so we can see the first previews of RTL format on huggingface docs.

No worries brother. I just checked the commit in your fork and I very much like the tone of your translation. It sounds natural and shows that you've spent time to refine it--that's valuable work. My PRs have "first draft" in the name for this very reason. We need to collaborate more swiftly by sharing drafts of our work and learning from each other. With smaller PRs and and more frequent PR reviews, we can keep our working trees updated with the latest changes. I will spend some time tonight, to create a second draft by mixing mine with yours and I will also add that empty file we need for chapter 1. I'll mention you in the PR when I'm done so you can review the result and it can be merged.

**Comment by jowharshamshiri** [comment]
> @jowharshamshiri I am not sure how the translated numbers will be shown in the docs and which is the best approach.
> 
> ```
> - title: 0. نصب
>   sections:
>   - local: chapter0/1
>     title: مقدمه
> ```
> 
> Or?
> 
> ```
> - title: ۰. نصب
>   sections:
>   - local: chapter0/1
>     title: مقدمه
> ```

I think this works, but let's wait and see the live previews.

```

- title: ۲- بکارگیری ترنسفورمرهای هاگینگ‌فیس
  sections:
  - local: chapter2/1

```

**Comment by lewtun** [comment]
Hi @jowharshamshiri I suppose one idea would be to include a `TRANSLATING.txt` file in `chapters/fa` - it can't be a `.md` file because `doc-builder` will try to build it as part of the website. 

This way we keep the glossary clean on the website. What do you think?

**Comment by jowharshamshiri** [comment]
> Hi @jowharshamshiri I suppose one idea would be to include a `TRANSLATING.txt` file in `chapters/fa` - it can't be a `.md` file because `doc-builder` will try to build it as part of the website.
> 
> This way we keep the glossary clean on the website. What do you think?

That's perfect thank you. Will do.

**Comment by schoobani** [comment]
@jowharshamshiri 
Added Chapter1 --> `1.mdx` and `2.mdx` and they are ready for the peer reviewing.
https://github.com/schoobani/course/tree/main/chapters/fa/chapter1

I used brackets to indicate phrases which doesn't appear to be in the glossary and I think they will appear again in the text. here is a list of them and my initial translations for them.

- Fine-tune (فاین‌تون)
- Research Engineer (مهندس محقق)
- Naive Base (بیز ساده)
- Task (مسئله)
- Demo (دمو)
- Prompt (متن اولیه‌ی پرامپت)

I also faced the issue that beginning an RTL row with an english phrase destroys the whole line in github. I guess it would be the same in huggingface and we should somehow ignore doing so.

**Comment by schoobani** [comment]
@jowharshamshiri @lewtun 
I think we should consider using another font for the persian/arabic in huggingface. 
In my opinion, current used font is very difficult to read and follow.

**Comment by jowharshamshiri** [comment]
> @jowharshamshiri Added Chapter1 --> `1.mdx` and `2.mdx` and they are ready for the peer reviewing. https://github.com/schoobani/course/tree/main/chapters/fa/chapter1
> 
> I used brackets to indicate phrases which doesn't appear to be in the glossary and I think they will appear again in the text. here is a list of them and my initial translations for them.
> 
>     * Fine-tune (فاین‌تون)
> 
>     * Research Engineer (مهندس محقق)
> 
>     * Naive Base (بیز ساده)
> 
>     * Task (مسئله)
> 
>     * Demo (دمو)
> 
>     * Prompt (متن اولیه‌ی پرامپت)
> 
> 
> I also faced the issue that beginning an RTL row with an english phrase destroys the whole line in github. I guess it would be the same in huggingface and we should somehow ignore doing so.

Hey @schoobani. Great progress! Was planning on doing this tonight but have been swamped lately and honestly am having trouble keeping my eyes open. Will get back to you as soon as I can. :)

**Comment by kambizG** [comment]
Hi, I'd like to contribute in this project. Please, let me know what I can do? I saw someone has already taken chapter 1. Can I start with Ch2?

**Comment by schoobani** [comment]
> 

Hi @kambizG! chapter1 is work in progress by me and chapter2 will be done by @jowharshamshiri. 
We can ask @lewtun to assign chapter3 **Fine-tuning a pretrained model** to you.

**Comment by jowharshamshiri** [comment]
Hello and very happy to have you joining us @kambizG jan. Thanks for your precious time. Please have a look at the glossary before you start and tell us how you think we could do better.

**Comment by hamedonline** [comment]
Hi everyone!
Great to see Persian community here 🤗. I would also like to contribute by taking over chapter4, if you will. Let's get it done ;)

**Comment by jowharshamshiri** [comment]
> Hi everyone! Great to see Persian community here hugs. I would also like to contribute by taking over chapter4, if you will. Let's get it done ;)

Hi @hamedonline jan. Super happy to have you join us. Thanks for donating your time. Please checkout the glossary, it is a work in progress and your input is very much appreciated.

@lewtun I added Hamed's name to chapter 4. 
@kambizG Please confirm if you'd like to work on chapter 3.

**Comment by jowharshamshiri** [comment]
> 



> @jowharshamshiri Added Chapter1 --> `1.mdx` and `2.mdx` and they are ready for the peer reviewing. https://github.com/schoobani/course/tree/main/chapters/fa/chapter1
> 
> I used brackets to indicate phrases which doesn't appear to be in the glossary and I think they will appear again in the text. here is a list of them and my initial translations for them.
> 
>     * Fine-tune (فاین‌تون)
> 
>     * Research Engineer (مهندس محقق)
> 
>     * Naive Base (بیز ساده)
> 
>     * Task (مسئله)
> 
>     * Demo (دمو)
> 
>     * Prompt (متن اولیه‌ی پرامپت)
> 
> 
> I also faced the issue that beginning an RTL row with an english phrase destroys the whole line in github. I guess it would be the same in huggingface and we should somehow ignore doing so.

Hey @schoobani, sorry for the late reply, literally was half-awake when I wrote the last message. Please submit these pages in a PR, I think the best place for this discussion is in the PR page while we review the work. Here are my notes on the two pages. They are ordered as they occurred to me in reading. I welcome further input from you and our new collaborators @kambizG and @hamedonline on these points.

PAGE 1:

1. Refer to the glossary on the use of the huggingface emoji.

2. Library names are not brand names, nor are they concepts, so no need to transliterate. They are shorter namespaces so just use the English equivalent.
For example we will transliterate 'Tokenizer' when it is used to indicate a concept but not when it is the name of a library.

3. I don't think we should transliterate the word Fine-tune.

4. I don't think we should transliterate the word Demo.

5. You need to surround the entire doc with a <dev dir="rtl">, also for lists separately. They render in LTR with the current version of Huggingface docs if you don't do this and the preview will look ugly. Refer to our previously submitted pages.

6. I defer to our new collaborators on whether we need to add context info about AGI. That info is always a google search away and the hard and fast rule is to not include ANY new contextual info. Refer to point # in the notes on translating at the bottom of the glossary. I'll soon be moving those out to a text file, per your suggestion.

7. I think when the text talks about "robot immortality" it doesn't mean that robots will become immortal--they already are. It's talking about humans and machines merging and humans acheieving immortality through that. Also does scale up translate to the Persian "افزایش مقیاس"? Again I defer to our new collaborators.

8. 'اولین گام‌های آغازین' is redundant.

9. A couple of uses of 'Yek' as a substitute for English indefinite articles.

10. I don't think to democratize translates well to the Persian "دموکراتیک کردن". The Persian has solely political connotations to me, but its use in the English text is not political at all. I defer to our new collaborators on a better equivalent.

11. BigScience is a brand name, no need to transliterate.

12. Do we have هم-نویسنده‌ in Persian?

13. I think your choice of تابع is better than the glossary's. Please include that update to the glossary in your PR.

14. I defer to our new collaborators on whether we should transliterate encoder and decoder or use the Persian equivalents رمزنگار and رمزگشا. Do we know what is used in Persian textbooks? and do we like that?

PAGE 2

15. Again hard-and-fast rule is not to expand the acronym NLP when the original text hasn't. There is some gray area here, as in the case of ML that we talk about in the glossary(Acronyms used for convenience in English). With some nuance I think expanding it into full Persian form is acceptable in some places in Page 2. If you choose to do that, please choose only one form, no need to include the acronym in ellipses.

16. Persian sentences follow the format Subject-Object-Verb. That means it is generally expected that the sentences end in a verb. Again there is a lot of gray area here and we need to apply nuance. But try to keep that in mind. For example maybe don't end sentences in "هستند یا نه". I say maybe bc if you do, that is your creative choice and I respect that.

17. I propose the use of "تشخیص نوع" in place of "دسته‌بندی" in Page 2, the verb 'to classify' is not used as a technical term there.

18. Is "بیان کردن" our best equivalent for 'represent'?

19. Chapter is فصل and section is بخش. The glossary is your friend.

**Comment by kambizG** [comment]
> > 
> 
> Hi @kambizG! chapter1 is work in progress by me and chapter2 will be done by @jowharshamshiri. We can ask @lewtun to assign chapter3 **Fine-tuning a pretrained model** to you.

Great! then I'll start Ch3 as soon as possible. Just a quick question, since this is my first time doing such contributions. I'm gonna do as the following: 1) clone, 2) translate, 3) push 

is that right?

**Comment by kambizG** [comment]
> > Hi everyone! Great to see Persian community here hugs. I would also like to contribute by taking over chapter4, if you will. Let's get it done ;)
> 
> Hi @hamedonline jan. Super happy to have you join us. Thanks for donating your time. Please checkout the glossary, it is a work in progress and your input is very much appreciated.
> 
> @lewtun I added Hamed's name to chapter 4. @kambizG Please confirm if you'd like to work on chapter 3.

@jowharshamshiri, yes I'll start working. This is my first such contribution and I need to make sure I'm doing right. First, Should I clone the main branch? Second, is this right to do as follows: 1)clone, 2)translate, and 3) push?

**Comment by jowharshamshiri** [comment]
> > > Hi everyone! Great to see Persian community here hugs. I would also like to contribute by taking over chapter4, if you will. Let's get it done ;)
> > 
> > 
> > Hi @hamedonline jan. Super happy to have you join us. Thanks for donating your time. Please checkout the glossary, it is a work in progress and your input is very much appreciated.
> > @lewtun I added Hamed's name to chapter 4. @kambizG Please confirm if you'd like to work on chapter 3.
> 
> @jowharshamshiri, yes I'll start working. This is my first such contribution and I need to make sure I'm doing right. First, Should I clone the main branch? Second, is this right to do as follows: 1)clone, 2)translate, and 3) push?

We are lucky to have you. Yes, that is about it. You ofc already know this stuff but just to be complete: 

1. Please fork the [Hugging Face course](https://github.com/huggingface/course) to your profile.
2. Clone your fork to your local machine.
3. Use this issue page for general discussion on word choices and whatnot.
4. When you have a page done commit back to your fork and open a PR for that page on the Hugging Face course repo.
5. Ask someone to help you review the page there. Commit back to your fork the changes and they will automatically be appended to the PR. If you have updates to the glossary mention them on this issue page so we can all apply the changes retroactively to our sections.
6. When done with the review, ask @lewtun to merge.

**Comment by jowharshamshiri** [comment]
@lewtun I added @kambizG 's name to Chapter 3.

**Comment by hamedonline** [comment]
Thank you @schoobani for the draft and thanks to @jowharshamshiri for extensive suggestions. I'd also like to submit a few hints regarding the discussed lines:
1. I guess we'd better check out the latest official Persian writing guide by 'فرهنگستان زبان و ادب فارسی' as well. Please note that **I'm not advocating for them at all**. It's just there are some changes compared to what my generation learned in high school, and skimming through that guide may help us get a clear mind about the writing style we choose to follow. For instance:
Nowadays, 'کلمهٔ فارسی' is preferred to 'کلمه‌ی فارسی'; contrary to what they taught us back then!
This is the link to latest version of the guide, as of now: https://apll.ir/wp-content/uploads/2018/10/D-1394.pdf
2. I kind of like the idea of writing English technical equivalent word besides the (common in literature) Persian translation; for example: رمزگشا (decoder).
This way we wouldn't worry about someone not being familiar with Persian form of the technical word.
3. Sometimes there are no good one-word translation equivalents for technical glossary words. Words such as 'Production', 'Inference' could be very misleading if translated literally. I think @schoobani did a great job on referring to 'Production' as 'استقرار نرم‌افزار'.
4. We definitely need to add more words to glossary as we continue and discuss the better representative translations. Based on this and previous point, here are my suggestions for some important technical words:
Demo -> نسخه نمایشی
Inference -> اجرا روی داده نمونه
Production -> استقرار / نرم‌افزار مستقر (depending on the sentence)
Fine-tune -> We use 'کوک کردن' literally in Persian for the translation, but as we may all agree this might not be a good translation. So why not use a more descriptive translation in such cases to convey the actual meaning? For example:
Fine-tuning a pre-trained model -> بهبود سازگاری مدل ازپیش تعلیم‌دیده (با داده‌های جدید)
to democratize machine learning for everyone -> تسهیل دسترسی به یادگیری ماشین برای همگان

**Comment by jowharshamshiri** [comment]
> Thank you @schoobani for the draft and thanks to @jowharshamshiri for extensive suggestions. I'd also like to submit a few hints regarding the discussed lines:
> 
>     1. I guess we'd better check out the latest official Persian writing guide by 'فرهنگستان زبان و ادب فارسی' as well. Please note that **I'm not advocating for them at all**. It's just there are some changes compared to what my generation learned in high school, and skimming through that guide may help us get a clear mind about the writing style we choose to follow. For instance:
>        Nowadays, 'کلمهٔ فارسی' is preferred to 'کلمه‌ی فارسی'; contrary to what they taught us back then!
>        This is the link to latest version of the guide, as of now: https://apll.ir/wp-content/uploads/2018/10/D-1394.pdf
> 
>     2. I kind of like the idea of writing English technical equivalent word besides the (common in literature) Persian translation; for example: رمزگشا (decoder).
>        This way we wouldn't worry about someone not being familiar with Persian form of the technical word.
> 
>     3. Sometimes there are no good one-word translation equivalents for technical glossary words. Words such as 'Production', 'Inference' could be very misleading if translated literally. I think @schoobani did a great job on referring to 'Production' as 'استقرار نرم‌افزار'.
> 
>     4. We definitely need to add more words to glossary as we continue and discuss the better representative translations. Based on this and previous point, here are my suggestions for some important technical words:
>        Demo -> نسخه نمایشی
>        Inference -> اجرا روی داده نمونه
>        Production -> استقرار / نرم‌افزار مستقر (depending on the sentence)
>        Fine-tune -> We use 'کوک کردن' literally in Persian for the translation, but as we may all agree this might not be a good translation. So why not use a more descriptive translation in such cases to convey the actual meaning? For example:
>        Fine-tuning a pre-trained model -> بهبود سازگاری مدل ازپیش تعلیم‌دیده (با داده‌های جدید)
>        to democratize machine learning for everyone -> تسهیل دسترسی به یادگیری ماشین برای همگان

Thank YOU for being amazing: on point 1, we were with you before you were even here! [see this comment.](https://github.com/huggingface/course/issues/50#issuecomment-1083672923) I didn't know they changed that convention back! I'll be sure to retroactively apply that to my pages in the next PR.

I am generally against inventing new words when we could just use or introduce transliterated loan words that are just as acceptable and communicate the message better especially since they make it easier for the Persian-speakers to search for the concept online and access the wealth of knowledge available there--a privilege that the English-speakers enjoy. We don't want to trap them in the bubble of information available only in Persian. I didn't know if that applied here with the case of "رمزگشا" bc if the word is accepted in Persian textbooks and widely used in Iranian industry there is no bubble to speak of. I unfortunately had to drop out of uni before I got to these concepts and defer to those of us with formal training. So on point 2, again I agree but maybe we could do it in footnotes? or endnotes in the "واژه نامه"? Your call. I only want us to avoid ellipses bc English characters in Persian text are a nuisance and break the flow of Persian. [See point 2 in this comment.](https://github.com/huggingface/course/issues/50#issuecomment-1086498256)

On point 3, I agree completely and yes @schoobani did a great job. Please update the glossary as you see fit.

On point 4, I again agree with all of your thoughts. I actually LOVE "کوک کردن". Missing piece of jigsaw falling perfectly into place. This is perfect use of words already available in Persian--it needs no extra explanation and is very efficient in communicating the message. Now that I heard it, I think we'd be wrong not to use it. Also agree with what you said about more descriptive multi-word translations, that is also a totally viable strategy: I like your suggestion for 'democratize'.

**Comment by kambizG** [comment]
> @lewtun I added @kambizG 's name to Chapter 3.

I started working. Try to finish intro by tomorrow.

**Comment by kambizG** [comment]
Hi, I finished Ch3-intro. Can someone please review my work? 

Here is my proposal for some new terms that I couldn't find in the glossary:

keras -> کراس
fine-tune -> بازتنظیم
prediction -> پیش بینی
training loop -> چرخه تعلیم
pre-trained -> پیش تعلیم

**Comment by jowharshamshiri** [comment]
Hi everyone. Please review page2 of chapter2. I have too many additions to the glossary to list them here. ❤️❤️❤️

https://github.com/huggingface/course/pull/129

**Comment by hamedonline** [comment]
Great contributions, thanks everyone 🙌.

I noticed that glossary will be a work in progress in which we might have different opinions on the most-fitting words as we go further. And soon it would result in inconsistent translations if we don't decide on final suitable words in glossary.
Therefore I decided to make a spreadsheet with multiple translation suggestion spots for each word. This would make new suggestions easier to track and keep them all in one place. Later on, the concluded (and agreed upon) translations could be added to glossary by making a PR (I believe the glossary should be the final map we refer to each time, not a suggestions pot). **This would make it far-easier for us and future contributors to provide more uniform translations.**

Please let me know what you think of this. Here's the link to online spreadsheet in which we can all add newly encountered words and propose our suggestions:
https://docs.google.com/spreadsheets/d/1HjtNTPiMfZmJkIp_0RP0SmVrbVCWP8vaCcT-2VbLbM0/edit?usp=sharing

If you all agree, I'll make the spreadsheet editable for you so that we can accelerate the process.

**Comment by jowharshamshiri** [comment]
> Great contributions, thanks everyone raised_hands.
> 
> I noticed that glossary will be a work in progress in which we might have different opinions on the most-fitting words as we go further. And soon it would result in inconsistent translations if we don't decide on final suitable words in glossary. Therefore I decided to make a spreadsheet with multiple translation suggestion spots for each word. This would make new suggestions easier to track and keep them all in one place. Later on, the concluded (and agreed upon) translations could be added to glossary by making a PR (I believe the glossary should be the final map we refer to each time, not a suggestions pot). **This would make it far-easier for us and future contributors to provide more uniform translations.**
> 
> Please let me know what you think of this. Here's the link to online spreadsheet in which we can all add newly encountered words and propose our suggestions: https://docs.google.com/spreadsheets/d/1HjtNTPiMfZmJkIp_0RP0SmVrbVCWP8vaCcT-2VbLbM0/edit?usp=sharing
> 
> If you all agree, I'll make the spreadsheet editable for you so that we can accelerate the process.
> 
> P.S. The spreadsheet contains the words discussed so far; it lacks the just-new additional ones from @jowharshamshiri 's comment above.

Hi Hamed Jan. We originally created the glossary file to serve this very purpose, it does not exist in the English version. Over at the Italian translation they are doing the same thing. So ofc thank you for the initiative and I know it will be useful, but we need to keep in sync by fetching and contributing to the glossary file, we need a single source of truth.

Also in relation to the PR issue, please take a look at these PRs in the other translation for example:

https://github.com/huggingface/course/pull/103
https://github.com/huggingface/course/pull/102
https://github.com/huggingface/course/pull/125

There is also the github workflow issue. There is in place a CI mechanism whereby we can preview our work as it will appear with full styles and everything. It only lives as long as the PR is open and is a great way to collaborate and find problems. We don't have access to that in local forks. The PR thread is the place for discussion and when we always fetch from upstream we have the latest changes in the glossary too!

**Comment by jowharshamshiri** [comment]
> 

Hi @kambizG jan, 

Great job and thank you. 

In case of "Keras", Can we add diacritics for to ensure our readers get the correct pronunciation? Please see the last guideline in the new TRANSLATING file in my PR.

I vote for "کوک کردن" for fine-tune. Guideline no. 8 in the old glossary and the new TRANSLATING file applies here. Do we consider this to be a technical word? If not, why not use both.

In case of "prediction", I didn't know what would be cool, bc I felt "prediction" didn't really apply to the classification task in my page, so I went with "تولید خروجی", I defer to you, and will fix my PR shortly.

Also in case of "pre-trained", maybe we can go with @hamedonline 's suggestion and make it a multi-word phrase? like از پیش تعلیم دیده and leave it to the individual collaborator to find the best fit for their particular sentence?

**Comment by hamedonline** [comment]
Ok dear friends, Chapter4/Page1 is ready for review:
https://github.com/hamedonline/huggingface-course/blob/main/chapters/fa/chapter4/1.mdx
Please let me know your thoughts.

New word proposals / edits for glossary:

- Dataset -> مجموعه داده
- Community -> انجمن کاربران
- Reproducibility -> بازتولید
- Versioning -> نسخه‌بندی
- Tokenizer -> توکِن‌ساز
- Feature (as in 'dataset features') -> ویژگی
- High-Dimensional -> بعد بالا
- Multi-Dimensional -> چند بعدی
- Loss Function -> تابع هزینه
- Activation Layer -> لایه فعال‌سازی
- Activation Function -> تابع فعال‌سازی
- Cross Entropy -> آنتروپی متقابل

A general rule could be we consider creating a footnote for unfamiliar Persian words, and make a reference to the main English technical word there.

**Comment by hamedonline** [comment]
> Hi, I finished Ch3-intro. Can someone please review my work? https://github.com/kambizG/course/tree/main/chapters/fa/chapter3
> 
> Here is my proposal for some new terms that I couldn't find in the glossary:
> 
> keras -> کراس fine-tune -> بازتنظیم prediction -> پیش بینی training loop -> چرخه تعلیم pre-trained -> پیش تعلیم

Dear @kambizG thanks for the great work.
I just noticed the lack of نیم‌فاصله in some words. That would be awesome if you consider that. For instance:

اما چگونه میتوانید یک مدل پیش تعلیم را خودتان بازتنظیم کنید؟
Changes to:
اما چگونه می‌توانید یک مدل ازپیش‌تعلیم‌دیده را خودتان بازتنظیم کنید؟
Or even could change to:
اما چگونه می‌توانید یک مدل ازپیش‌تعلیم‌دیده را خودتان با داده‌های جدید بهبود دهید؟

**Comment by jowharshamshiri** [comment]
> 

Hi Hamed Jan,

On "dataset", I vote to transliterate. This is an accepted word in the industry from my vantage point. Not once in years have I said "majmoue-dade" to any colleague. Dataset we use all the time and it is a first-class Persian word, even if it is a loan word.

On Hub, if it is not The Hugging Face Hub(which I think is also referred to the Model Hub elsewhere), I agree with your choice.

On Model Hub, I think we should make it easy for people to google this. Also this is sort of a brand for Hugging Face I think, so I vote we don't even transliterate and use the original English. My second choice would be transliteration.

On Community, I vote we go with "جامعه کاربران"، "anjoman" seems a little too forum-y to me.

On Reproducibility, I think you mean قابل باز تولید, I agree.

On versioning I agree.

On tokenizer, this is exactly the kind of thing we'd want to transliterate. It is a concept and we need people just starting out to immediately come up with the closest English spelling and search for it online.

On features in datasets, and also in models, I vote for transliteration, but defer to you. If we do choose "ویژگی " please also come up with a better word that what we have, for attribute, bc it clashes with this one.

On High-Dimensional, I went with "چندین بعدی", I think yours is better.

On Multi-Dimensional I agree.

On Loss Function, thank you for illuminating me, I did not know what the Persian equivalent for this was, and had to resort to transliteration which felt weird.

On Activation Layer, I agree. Should it end in "ی" or not please make this clear. We have this problem with یادگیری ماشین vs یادگیری ماشینی too.

On Activation Function, I agree. Please make the same clarification as above.

On Cross Entropy, again thanks for this. Had to transliterate. Felt weird.

A lot of new words in the glossary, please also tell us what you think about them. I will be doing a new PR soon to include our new choices whatever they are in my pages.

**Comment by jowharshamshiri** [comment]
> Ok dear friends, Chapter4/Page1 is ready for review: https://github.com/hamedonline/huggingface-course/blob/main/chapters/fa/chapter4/1.mdx Please let me know your thoughts.
> 
> New word proposals / edits for glossary:
> 
>     * Dataset -> مجموعه داده
> 
>     * Hub -> پایگاه
> 
>     * Model Hub -> پایگاه مدل
> 
>     * Community -> انجمن کاربران
> 
>     * Reproducibility -> بازتولید
> 
>     * Versioning -> نسخه‌بندی
> 
>     * Tokenizer -> توکِن‌ساز
> 
>     * Feature (as in 'dataset features') -> ویژگی
> 
>     * High-Dimensional -> بعد بالا
> 
>     * Multi-Dimensional -> چند بعدی
> 
>     * Loss Function -> تابع هزینه
> 
>     * Activation Layer -> لایه فعال‌سازی
> 
>     * Activation Function -> تابع فعال‌سازی
> 
>     * Cross Entropy -> آنتروپی متقابل
> 
> 
> Please also let me know about this suggestions and I'll update the glossary accordingly.
> 
> A general rule could be we consider creating a footnote for unfamiliar Persian words, and make a reference to the main English technical word there.

> 

Hi @hamedonline  Jan,

On "dataset", I vote to transliterate. This is an accepted word in the industry from my vantage point. Not once in years have I said "majmoue-dade" to any colleague. Dataset we use all the time and it is a first-class Persian word, even if it is a loan word.

On Hub, if it is not The Hugging Face Hub(which I think is also referred to the Model Hub elsewhere), I agree with your choice.

On Model Hub, I think we should make it easy for people to google this. Also this is sort of a brand for Hugging Face I think, so I vote we don't even transliterate and use the original English. My second choice would be transliteration.

On Community, I vote we go with "جامعه کاربران"، "anjoman" seems a little too forum-y to me.

On Reproducibility, I think you mean قابل باز تولید, I agree.

On versioning I agree.

On tokenizer, this is exactly the kind of thing we'd want to transliterate. It is a concept and we need people just starting out to immediately come up with the closest English spelling and search for it online.

On features in datasets, and also in models, I vote for transliteration, but defer to you. If we do choose "ویژگی " please also come up with a better word that what we have, for attribute, bc it clashes with this one.

On High-Dimensional, I went with "چندین بعدی", I think yours is better.

On Multi-Dimensional I agree.

On Loss Function, thank you for illuminating me, I did not know what the Persian equivalent for this was, and had to resort to transliteration which felt weird.

On Activation Layer, I agree. Should it end in "ی" or not please make this clear. We have this problem with یادگیری ماشین vs یادگیری ماشینی too.

On Activation Function, I agree. Please make the same clarification as above.

On Cross Entropy, again thanks for this. Had to transliterate. Felt weird.

A lot of new words in the glossary, please also tell us what you think about them. I will be doing a new PR soon to include our new choices whatever they are in my pages.

**Comment by jowharshamshiri** [comment]
I updated the issue prompt up there with new information on where the important files are and steps for contributing.

**Comment by kambizG** [comment]
Hello friends, 

The first draft of Ch3 - 2.mdx is finally done before the weekend. I'll go through the chapter during the weekend and ask for a reviewer's help ASAP. 

Here is my new proposal for the glossary:

cache -> ذخیره کردن
object -> شئ
integer -> مقدار عددی صحیح
pre-processing -> پیش پردازش
paraphrases -> جملات متناظر
benchmark -> محک
items -> اقلام
argument -> آرگومان
padding -> هم طول سازی
instantiate -> معرفی کردن

Also, just a note: I'm getting a lot of help from www.abadis.ir.

ps: @hamedonline jan, thanks for reviewing the 1.mdx and also for setting up the google sheet, that's very helpful.

**Comment by jowharshamshiri** [comment]
> Hello friends,
> 
> The first draft of Ch3 - 2.mdx is finally done before the weekend. I'll go through the chapter during the weekend and ask for a reviewer's help ASAP.
> 
> Here is my new proposal for the glossary:
> 
> cache -> ذخیره کردن object -> شئ integer -> مقدار عددی صحیح pre-processing -> پیش پردازش paraphrases -> جملات متناظر benchmark -> محک items -> اقلام argument -> آرگومان padding -> هم طول سازی instantiate -> معرفی کردن
> 
> Also, just a note: I'm getting a lot of help from [www.abadis.ir](http://www.abadis.ir).
> 
> ps: @hamedonline jan, thanks for reviewing the 1.mdx and also for setting up the google sheet, that's very helpful.

Hi @kambizG  jan,

On cache, I see now that I did have this on ch2/p2 and used the same equivalent as you, but forgot to add it to the glossary. If you want I can add it to my PR.

On object, I agree. New word.

On Integer, I just went with "عدد صحیح" and if that is ok with you. Already in the G.

On pre-processing, I went with "پیش‌پردازش"(with a half-space) bc it seemed like a single semantic unit to me. Do you think I should change it? Same thing with post-processing, "پس‌پردازش". Both already in G.

On paraphrases, I agree. New word.

On benchmark, I like "معیار" more, but defer to you. New word.

On items, I agree. New word.

On argument, I agree. Already in G.

On padding, I defer to you. New word.

On instantiate already in G with "ساختن". I think that sounds more natural.

Please check pages 22-24 of the [style guide](https://apll.ir/wp-content/uploads/2018/10/D-1394.pdf) on prefixes and suffixes and page 10 for spacing.

While I have your attention, will you please be kind enough to review the [second draft of my pages](https://github.com/huggingface/course/pull/139)? I'd love to hear what you think about my pages. It helps if you catch my problems early so I won't have to edit all my pages at once.

**Comment by jowharshamshiri** [comment]
Hi @hamedonline jan,

Please review the [first draft of Chapter2/Page3](https://github.com/huggingface/course/pull/140/files).

**Comment by hamedonline** [comment]
> Hi @hamedonline jan,
> 
> Please review the [first draft of Chapter2/Page3](https://github.com/huggingface/course/pull/140/files).

Great work @jowharshamshiri aziz ❤️ 
I'll check it out tonight for sure.



> 1. Take this sentence. Should اشتراک and گذاشته be separated by a half-space per our last review? Seems weird.
>    . ...را به اشتراک گذاشته...

This is a good example to discuss and figure out how we should treat these cases. On Ch4P1 I had a similar situation (you most probably noticed it). The phrase was "به اشتراک گذاشتن مدل در هاب...". I treated it with full space and as you just said, it seemed weird with half-space. On the other hand, the case in which I applied half-space was with "اشتراک‌گذاری" which feels as an entirety to me, and we borrowed it from style guide which was full of half-spaced "گذاری" instances.

Now, let's imagine we are doing a translation job from Persian to English. Having half-spaces considered in the text, is a bliss in such occasions, since it allows the translator understand message of lines more easily. Take a look at these three examples:

1. with half-space: "فایل به اشتراک‌گذاشته‌شده" -> "the shared file"
2. with full-space: "فایل به اشتراک گذاشته شده" -> "file has been shared"
3. with full-space: "فایلی که به اشتراک گذاشته شده" -> "the file that has been shared"

Now, I think the half-space in first example is utilized to put emphasis on the 'objective' nature of "shared". Semantically this is right, even more rightful when written as: "فایل به‌اشتراک‌گذاشته‌شده".
But writing in that style feels weird, doesn't it?

Anyhow, I believe "به اشتراک گذاشتن" with full-space and "به اشتراک‌گذاری" with half-space before -gozari- part is the way to go, and your choice of "...را به اشتراک گذاشته..." is right; and let's not forget we don't have an 'objective' or "مفعولی" situation here, it is rather "فاعلی".

> 2\. Both me and @kambizG jan, previously did not distinguish btw equivalents for "cache" and "save". I have a section on how caching of models happens immediately followed by a section on saving the models, in the new page. How do you guys think we should proceed?

I think since "caching" and "saving" have different meanings, we should definitely think of distinguishable Persian replacements. Will do some digging.

> 3\. I previously asked about the accepted equivalent for "Encoder" and "Decoder" while reviewing @schoobani jan's work. I have the verb "encode" here, but it's only used to mean we assign numbers to things, for which "رمزنگاری" doesn't really work. How do you guys think we should proceed? "کد گذاری" or "کد شدن" maybe?

I had this mentioned actually in google spreadsheet. If we use رمز in here for encoder/decoder, what equivalent should be used when encountered with "cypher" or "decipher"?
Actually "کدگذار" and "کدگشا" seem more proper to me as I dive deeper.
Or maybe we could just transliterate and go with انکودر / دیکودر?

**Comment by jowharshamshiri** [comment]
> > Hi @hamedonline jan,
> > Please review the [first draft of Chapter2/Page3](https://github.com/huggingface/course/pull/140/files).
> 
> Great work @jowharshamshiri aziz heart I'll check it out tonight for sure.
> 
> > 1. Take this sentence. Should اشتراک and گذاشته be separated by a half-space per our last review? Seems weird.
> >    . ...را به اشتراک گذاشته...
> 
> This is a good example to discuss and figure out how we should treat these cases. On Ch4P1 I had a similar situation (you most probably noticed it). The phrase was "به اشتراک گذاشتن مدل در هاب...". I treated it with full space and as you just said, it seemed weird with half-space. On the other hand, the case in which I applied half-space was with "اشتراک‌گذاری" which feels as an entirety to me, and we borrowed it from style guide which was full of half-spaced "گذاری" instances.
> 
> Now, let's imagine we are doing a translation job from Persian to English. Having half-spaces considered in the text, is a bliss in such occasions, since it allows the translator understand message of lines more easily. Take a look at these three examples:
> 
>     1. with half-space: "فایل به اشتراک‌گذاشته‌شده" -> "the shared file"
> 
>     2. with full-space: "فایل به اشتراک گذاشته شده" -> "file has been shared"
> 
>     3. with full-space: "فایلی که به اشتراک گذاشته شده" -> "the file that has been shared"
> 
> 
> Now, I think the half-space in first example is utilized to put emphasis on the 'objective' nature of "shared". Semantically this is right, even more rightful when written as: "فایل به‌اشتراک‌گذاشته‌شده". But writing in that style feels weird, doesn't it?
> 
> Anyhow, I believe "به اشتراک گذاشتن" with full-space and "به اشتراک‌گذاری" with half-space before -gozari- part is the way to go, and your choice of "...را به اشتراک گذاشته..." is right; and let's not forget we don't have an 'objective' or "مفعولی" situation here, it is rather "فاعلی".
> 
> > 2. Both me and @kambizG jan, previously did not distinguish btw equivalents for "cache" and "save". I have a section on how caching of models happens immediately followed by a section on saving the models, in the new page. How do you guys think we should proceed?
> 
> I think since "caching" and "saving" have different meanings, we should definitely think of distinguishable Persian replacements. Will do some digging.
> 
> > 3. I previously asked about the accepted equivalent for "Encoder" and "Decoder" while reviewing @schoobani jan's work. I have the verb "encode" here, but it's only used to mean we assign numbers to things, for which "رمزنگاری" doesn't really work. How do you guys think we should proceed? "کد گذاری" or "کد شدن" maybe?
> 
> I had this mentioned actually in google spreadsheet. If we use رمز in here for encoder/decoder, what equivalent should be used when encountered with "cypher" or "decipher"? Actually "کدگذار" and "کدگشا" seem more proper to me as I dive deeper. Or maybe we could just transliterate and go with انکودر / دیکودر?

On 1, I agree let's not do example 1.
I also think eshterak-gozari does not warrant a half-space in any form, as evidenced by the fact that these two words form so many other patterns we use every day. Would also be hard to enforce in reviews and will confuse collaborators.

On 3, I vote we transliterate "Decoder" and "Encoder"(bc they're loanwords that stand for specific ML concepts), kod-gozari and kod-goshayi for the verbs encode and decode in benign non-cypher-y use(which sound natural to me), and ramz-gozari and ramz-goshayi for encrypt/decrypt(which is standard I think, not sure we even have them in this text tho).

**Comment by hamedonline** [comment]
> On 3, I vote we transliterate "Decoder" and "Encoder"(bc they're loanwords that stand for specific ML concepts), kod-gozari and kod-goshayi for the verbs encode and decode in benign non-cypher-y use(which sound natural to me), and ramz-gozari and ramz-goshayi for encrypt/decrypt(which is standard I think, not sure we even have them in this text tho).

Oh, I totally forgot encrypt / decrypt. There is that too :))
My vote is also transliteration on this one. Probably we could use "کدگذاری" somehow in possible scenarios like this (I made this example up btw):

"Encoder neural networks help to reduce dimensions by encoding the input."
"شبکه‌های عصبی از نوع انکودر، با کدگذاری ورودی به کاهش ابعاد کمک می‌کنند"

**Comment by kambizG** [comment]
Hi all, I'm done with the followings:

Ch3-P1 - first review applied:
https://github.com/kambizG/course/blob/main/chapters/fa/chapter3/1.mdx

Ch2-P2 - needs review (_**can someone review please?**_):
https://github.com/kambizG/course/blob/main/chapters/fa/chapter3/2.mdx

**Glossary:** 
----------------
cache -> ذخیره‌کردن
object -> شیء
integer -> مقدار صحیح
pre-processing -> پیش‌پردازش
paraphrases -> جملات متناظر
benchmark -> محک
items -> اقلام
argument -> آرگومان
padding -> هم‌طول‌سازی
instantiate -> تعریف کردن
sequential Model -> مدل ترتیبی
documentation -> مستندات
label -> برچسب
checkpoint -> چکپوینت

**Questions:**
----------------
The "en" sections of text that need to be in-line with "fa" text sometimes do not preserve the order. For instance:
[CLS] sentence1 [SEP] sentence2 [SEP]
is shown as: 
بنابراین می‌بینیم که مدل انتظار دارد وقتی که دو جمله داریم ورودی‌ها به صورت [CLS] sentence1 [SEP] sentence2 [SEP] باشند.

Any ideas?

**Comment by jowharshamshiri** [comment]
> Hi all, I'm done with the followings:
> 
> Ch3-P1 - first review applied: https://github.com/kambizG/course/blob/main/chapters/fa/chapter3/1.mdx
> 
> Ch3-P2 - needs review (_**can someone review please?**_): https://github.com/kambizG/course/blob/main/chapters/fa/chapter3/2.mdx
> ## **Glossary:**
> 
> cache -> ذخیره‌کردن object -> شیء integer -> مقدار صحیح pre-processing -> پیش‌پردازش paraphrases -> جملات متناظر benchmark -> محک items -> اقلام argument -> آرگومان padding -> هم‌طول‌سازی instantiate -> تعریف کردن sequential Model -> مدل ترتیبی documentation -> مستندات label -> برچسب checkpoint -> چکپوینت
> ## **Question:**
> 
>     1. The "en" sections of text that need to be in-line with "fa" text sometimes do not preserve the order. For instance:
>        [CLS] sentence1 [SEP] sentence2 [SEP]
>        is shown as:
>        بنابراین می‌بینیم که مدل انتظار دارد وقتی که دو جمله داریم ورودی‌ها به صورت [CLS] sentence1 [SEP] sentence2 [SEP] باشند.
> 
> 
> Any ideas?
> 
>     2. I suggest we create a channel somewhere (discord, slack, telegram, etc), where we can discuss some details? If you agree, just email me your phone numbers and I'll fix it.

Hi Kambiz jan,

On cache, @hamedonline suggested انبار کردن in his review of my page. It will be merged into the glossary once my PR is merged. That fit my use case fine, does it fit yours?

On label, I used "نام دسته" in a segment on classification. Tell me if you think I should change it to برچسب دسته, personally I think برچسب is a little too literal a translation. 

On checkpoint, I used نقطه تعلیم. Initially it seemed weird to me but I didn't have a better alternative. But having used it in multiple places now, it's grown on me. It really clicked for me here in this sentence:

```
این فایل حاوی مقادیری متادیتا، مانند نقطه مادر این نقطه تعلیم و نسخه‌‌ کتابخانه ترنسفورمرهای هاگینگ‌فِیس که آخرین بار برای ذخیره‌سازی این نقطه به کار رفته است، می‌باشد.
```

What do you think?

I'm not sure I understand the question you posed. Do you mean that when you mix English text with Persian it causes Github or maybe your editor to not render it correctly? If not ignore the rest of this. If the order is correct, regardless of what is shown, it renders correctly once it is deployed on Hugging Face docs.
Also when you have ellipses(like in function names) or maybe even brackets like your case here in that English phrase, and put the whole doc in an RTL div, it causes the ellipses to render AFTER the phrase which is not what you want. That's why in T/point13 we talk about surrounding those phrases in a span. Sorry if I didn't understand this correctly.

**Comment by kambizG** [comment]
> > Hi all, I'm done with the followings:
> > Ch3-P1 - first review applied: https://github.com/kambizG/course/blob/main/chapters/fa/chapter3/1.mdx
> > Ch3-P2 - needs review (_**can someone review please?**_): https://github.com/kambizG/course/blob/main/chapters/fa/chapter3/2.mdx
> > ## **Glossary:**
> > cache -> ذخیره‌کردن object -> شیء integer -> مقدار صحیح pre-processing -> پیش‌پردازش paraphrases -> جملات متناظر benchmark -> محک items -> اقلام argument -> آرگومان padding -> هم‌طول‌سازی instantiate -> تعریف کردن sequential Model -> مدل ترتیبی documentation -> مستندات label -> برچسب checkpoint -> چکپوینت
> > ## **Question:**
> > ```
> > 1. The "en" sections of text that need to be in-line with "fa" text sometimes do not preserve the order. For instance:
> >    [CLS] sentence1 [SEP] sentence2 [SEP]
> >    is shown as:
> >    بنابراین می‌بینیم که مدل انتظار دارد وقتی که دو جمله داریم ورودی‌ها به صورت [CLS] sentence1 [SEP] sentence2 [SEP] باشند.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Any ideas?
> > ```
> > 2. I suggest we create a channel somewhere (discord, slack, telegram, etc), where we can discuss some details? If you agree, just email me your phone numbers and I'll fix it.
> > ```
> 
> Hi Kambiz jan,
> 
> On cache, @hamedonline suggested انبار کردن in his review of my page. It will be merged into the glossary once my PR is merged. That fit my use case fine, does it fit yours?
> 
> On label, I used "نام دسته" in a segment on classification. Tell me if you think I should change it to برچسب دسته, personally I think برچسب is a little too literal a translation.
> 
> On checkpoint, I used نقطه تعلیم. Initially it seemed weird to me but I didn't have a better alternative. But having used it in multiple places now, it's grown on me. It really clicked for me here in this sentence:
> 
> ```
> این فایل حاوی مقادیری متادیتا، مانند نقطه مادر این نقطه تعلیم و نسخه‌‌ کتابخانه ترنسفورمرهای هاگینگ‌فِیس که آخرین بار برای ذخیره‌سازی این نقطه به کار رفته است، می‌باشد.
> ```
> 
> What do you think?
> 
> I'm not sure I understand the question you posed. Do you mean that when you mix English text with Persian it causes Github or maybe your editor to not render it correctly? If not ignore the rest of this. If the order is correct, regardless of what is shown, it renders correctly once it is deployed on Hugging Face docs. Also when you have ellipses(like in function names) or maybe even brackets like your case here in that English phrase, and put the whole doc in an RTL div, it causes the ellipses to render AFTER the phrase which is not what you want. That's why in T/point13 we talk about surrounding those phrases in a span. Sorry if I didn't understand this correctly.

Hi Bahram jan, 

**On Cache:** when I searched "Cache vs Storage/Save" it said that "Cache" is often used when we mean "Temporary Storage", which makes sense. For instance we always say "caching in RAM" and Not storing. Thus, I believe "انبار کردن" conducts more semantic weight towards "long term storage", rather than short. However, even "ذخیره سازی" is not a good choice as it somehow interferes with "Storage" or "Save". Instead, I'd rather suggest we either go with transliteration "کَش" or "ذخیره سازی موقت" or "انبار موقت".
 
On label, I believe we can even transliterate "لِیبِل" as it is first of all, a very common ML phrase and second, also commonly used in Persian. Otherwise, I believe I've seen "برچسب" in other technical CS translations to fa and would be nice to conform with the community.

On Checkpoint: I honestly didn't really understand and don't have very strong opinion. In my case it was used when the author wanted to say "when you decide to choose the model ..."!

Question 1: Sorry for the confusions. It's actually shown correct in here but not in my translation file. However, your interpretation is correct. When a section is a mixed of fa and en, sometimes the en part gets incorrectly re-ordered. For example "[CLS] sentence1 [SEP] sentence2 [SEP]" is shown as "sentence1 [SEP] sentence2 [SEP] [CLS]", the [CLS] goes to the end of the phrase. As you correctly mentioned this is also the case for parentheses, e.g., "map()" converts to "()map"!

**Comment by hamedonline** [comment]
Thank you @kambizG for your suggestions.
I think the addition of "موقت" term to **cache** translation makes total sense and both "انبار موقت" and "ذخیره‌سازی موقت" sound good to me. We may consider adding them both to glossary and use either one according to the vibe of translated sentence.

I'd also like to mention addition of a PR for a chapter4/page2 translation (as seen in above link). There are couple of proposals for glossary terms which I'll list again here for your valuable reviews (open to modify per your proposals):

Modification proposals:
- Instantiate -> ساختن / ایجاد نمونه شیء / نمونه‌سازی

New addition proposals:
- Pretrained Model(s) -> مدل(های) از پیش تعلیم دیده
- Downstream Library[^1] -> کتابخانه پایین‌دستی وابسته
- Mask Filling -> پر کردن جاهای خالی متن
- Interface -> رابط کاربری
- Bias -> سوگیری

[^1]: Please see this link for more info on this: https://reflectoring.io/upstream-downstream

**Comment by jowharshamshiri** [comment]
> > > Hi all, I'm done with the followings:
> > > Ch3-P1 - first review applied: https://github.com/kambizG/course/blob/main/chapters/fa/chapter3/1.mdx
> > > Ch3-P2 - needs review (_**can someone review please?**_): https://github.com/kambizG/course/blob/main/chapters/fa/chapter3/2.mdx
> > > ## **Glossary:**
> > > cache -> ذخیره‌کردن object -> شیء integer -> مقدار صحیح pre-processing -> پیش‌پردازش paraphrases -> جملات متناظر benchmark -> محک items -> اقلام argument -> آرگومان padding -> هم‌طول‌سازی instantiate -> تعریف کردن sequential Model -> مدل ترتیبی documentation -> مستندات label -> برچسب checkpoint -> چکپوینت
> > > ## **Question:**
> > > ```
> > > 1. The "en" sections of text that need to be in-line with "fa" text sometimes do not preserve the order. For instance:
> > >    [CLS] sentence1 [SEP] sentence2 [SEP]
> > >    is shown as:
> > >    بنابراین می‌بینیم که مدل انتظار دارد وقتی که دو جمله داریم ورودی‌ها به صورت [CLS] sentence1 [SEP] sentence2 [SEP] باشند.
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Any ideas?
> > > ```
> > > 2. I suggest we create a channel somewhere (discord, slack, telegram, etc), where we can discuss some details? If you agree, just email me your phone numbers and I'll fix it.
> > > ```
> > 
> > 
> > Hi Kambiz jan,
> > On cache, @hamedonline suggested انبار کردن in his review of my page. It will be merged into the glossary once my PR is merged. That fit my use case fine, does it fit yours?
> > On label, I used "نام دسته" in a segment on classification. Tell me if you think I should change it to برچسب دسته, personally I think برچسب is a little too literal a translation.
> > On checkpoint, I used نقطه تعلیم. Initially it seemed weird to me but I didn't have a better alternative. But having used it in multiple places now, it's grown on me. It really clicked for me here in this sentence:
> > ```
> > این فایل حاوی مقادیری متادیتا، مانند نقطه مادر این نقطه تعلیم و نسخه‌‌ کتابخانه ترنسفورمرهای هاگینگ‌فِیس که آخرین بار برای ذخیره‌سازی این نقطه به کار رفته است، می‌باشد.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > What do you think?
> > I'm not sure I understand the question you posed. Do you mean that when you mix English text with Persian it causes Github or maybe your editor to not render it correctly? If not ignore the rest of this. If the order is correct, regardless of what is shown, it renders correctly once it is deployed on Hugging Face docs. Also when you have ellipses(like in function names) or maybe even brackets like your case here in that English phrase, and put the whole doc in an RTL div, it causes the ellipses to render AFTER the phrase which is not what you want. That's why in T/point13 we talk about surrounding those phrases in a span. Sorry if I didn't understand this correctly.
> 
> Hi Bahram jan,
> 
> **On Cache:** when I searched "Cache vs Storage/Save" it said that "Cache" is often used when we mean "Temporary Storage", which makes sense. For instance we always say "caching in RAM" and Not storing. Thus, I believe "انبار کردن" conducts more semantic weight towards "long term storage", rather than short. However, even "ذخیره سازی" is not a good choice as it somehow interferes with "Storage" or "Save". Instead, I'd rather suggest we either go with transliteration "کَش" or "ذخیره سازی موقت" or "انبار موقت".
> 
> On label, I believe we can even transliterate "لِیبِل" as it is first of all, a very common ML phrase and second, also commonly used in Persian. Otherwise, I believe I've seen "برچسب" in other technical CS translations to fa and would be nice to conform with the community.
> 
> On Checkpoint: I honestly didn't really understand and don't have very strong opinion. In my case it was used when the author wanted to say "when you decide to choose the model ..."!
> 
> Question 1: Sorry for the confusions. It's actually shown correct in here but not in my translation file. However, your interpretation is correct. When a section is a mixed of fa and en, sometimes the en part gets incorrectly re-ordered. For example "[CLS] sentence1 [SEP] sentence2 [SEP]" is shown as "sentence1 [SEP] sentence2 [SEP] [CLS]", the [CLS] goes to the end of the phrase. As you correctly mentioned this is also the case for parentheses, e.g., "map()" converts to "()map"!

I defer to you on what we do with label and cache @kambizG jan.

**Comment by jowharshamshiri** [comment]
> Thank you dear @kambizG for your suggestions. I think the addition of "موقت" term to **cache** translation makes total sense and both "انبار موقت" and "ذخیره‌سازی موقت" sound good to me. We may consider adding them both to glossary and use either one according to the vibe of translated sentence.
> 
> I'd also like to mention addition of a PR for a chapter4/page2 translation (as seen in above link). There are couple of proposals for glossary terms which I'll list again here for your valuable reviews (open to modify per your proposals):
> 
> Modification proposals:
> 
>     * Instantiate -> ساختن / ایجاد نمونه شیء / نمونه‌سازی
> 
> 
> New addition proposals:
> 
>     * Pretrained Model(s) -> مدل(های) از پیش تعلیم دیده
> 
>     * Downstream Library[1](#user-content-fn-1-3f67abe9faa27628a7fbb8c4f6214837) -> کتابخانه پایین‌دستی وابسته
> 
>     * Mask Filling -> پر کردن جاهای خالی متن
> 
>     * Interface -> رابط کاربری
> 
>     * Bias -> سوگیری
> 
> 
> ## Footnotes
> 
>     1. Please see this link for more info on this: https://reflectoring.io/upstream-downstream [leftwards_arrow_with_hook](#user-content-fnref-1-3f67abe9faa27628a7fbb8c4f6214837)

Other than "نمونه‌سازی" I vote with you on all these @hamedonline jan. "ساختن نمونه" works why invent a new word. 

I especially love your equivalents for "bias" and "downstream library".

If you add "interface" to G, please use "user interface" to avoid duplication with API interfaces.

**Comment by jowharshamshiri** [comment]
> Hi folks, it's been really humbling to see how much care and effort you are all taking into translating the course into Persian ❤️ !
> 
> 
> 
> Regarding @kambizG's suggestion:
> 
> 
> 
> > 2\. I suggest we create a channel somewhere (discord, slack, telegram, etc), where we can discuss some details? If you agree, just email me your phone numbers and I'll fix it.
> 
> 
> 
> I've created a `persian-translation` channel on the Hugging Face Discord. To view it, you just need to activate it with an emoji on this message: https://discord.gg/hKnxnxUr
> 
> 
> 
> I'll add these details to the issue description - as a general rule, I think GitHub issues are great for discussing "big issues" while Discord is great for dynamic chats over smaller items. 
> 
> 
> 
> I also suggest linking to Discord messages in this issue wherever that's relevant, so people reading this can get the broader context 😄 

Folks, @schoobani jan and @hamedonline jan, please join the discord.

Easier to coordinate over there as me and @kambizG jan found out. We solved git issues over DM and I can't wait for us to hold a discussion over the equivalent for the word "padding" in main. The conversations there are more immediate and it's easier to clear up ambiguities in a timely manner like who is exactly reviewing my last PR now. Github really lacks DMs. We're waiting for you over there! ❤️❤️❤️

**Comment by jowharshamshiri** [comment]
> @jowharshamshiri It's perfect to use Discord but the Invite link is Invalid and I can't access it.

maybe try again? @hamedonline jan just joined with the same link. What do you mean invalid? does it take you to discord? if it does you need to react with an emoji to the message in the channel it takes you to, and wait a little for the admins to add you to our special discord.

**Comment by jowharshamshiri** [comment]
> ![Invite Invalid Error](https://user-images.githubusercontent.com/24176136/57620372-e1742c80-7588-11e9-9133-1d55e70b6877.png)
> 
> 

Weird! Gently pinging @lewtun on this.

**Comment by hamedonline** [comment]
@jowharshamshiri I got the same invalid invitation message yesterday. Have you already joined huggingface discord channel dear @schoobani?
If so, all you need to do is to go to course section and click on 🤗 **earth** emoji on #course-translations topic; after clicking that all sub-topics become activated and you can join conversations on #persian-translations.

**Comment by jowharshamshiri** [comment]
> @jowharshamshiri I got the same invalid invitation message yesterday. Have you already joined huggingface discord channel dear @schoobani?
> 
> If so, all you need to do is to go to course section and click on 🤗 **earth** emoji on #course-translations topic; after clicking that all sub-topics become activated and you can join conversations on #persian-translations.

I tap on the link and it takes me to the channel on discord but that might be bc I joined already. Anyways thanks to you for the instructions and I hope this works @schoobani and sorry for the unnecessary ping @lewtun.

**Comment by jowharshamshiri** [comment]
> @jowharshamshiri I got the same invalid invitation message yesterday. Have you already joined huggingface discord channel dear @schoobani? If so, all you need to do is to go to course section and click on 🤗 **earth** emoji on #course-translations topic; after clicking that all sub-topics become activated and you can join conversations on #persian-translations.

@schoobani jan here's the Hugging Face Discord link. https://discord.com/invite/JfAtkvEtRb
once there please follow hamed jan's instructions to find our channel.

**Comment by lewtun** [comment]
Hi folks, sorry the link expired - it seems you've sorted it out and as @jowharshamshiri suggests you can get the latest link via the Hub:

![Screen Shot 2022-05-18 at 21 19 32](https://user-images.githubusercontent.com/26859204/169139406-79d94323-73da-41c7-80c4-c1411088289d.png)
